pyomo/contrib/appsi/solvers/tests/test_gurobi_persistent.py

```diff
@@ -301,6 +392,21 @@ def test_nonconvex2(self):
         self.assertAlmostEqual(m.x.value, -0.3660254037844423, 2)
         self.assertAlmostEqual(m.y.value, -0.13397459621555508, 2)
 
+    def test_solution_number(self):
+        m = create_pmedian_model()
+        opt = Gurobi()
+        opt.gurobi_options['PoolSolutions'] = 3
+        opt.gurobi_options['PoolSearchMode'] = 2
+        res = opt.solve(m)
+        num_solutions = opt.get_model_attr('SolCount')
+        self.assertEqual(num_solutions, 3)
+        res.solution_loader.load_vars(solution_number=0)
+        self.assertAlmostEqual(pe.value(m.obj.expr), 6.431184939357673)
+        res.solution_loader.load_vars(solution_number=1)
+        self.assertAlmostEqual(pe.value(m.obj.expr), 6.584793218502477)
+        res.solution_loader.load_vars(solution_number=2)
+        self.assertAlmostEqual(pe.value(m.obj.expr), 6.592304628123309)
```

Comment:
How fragile will these tests be with different versions of Gurobi?  Is it sufficient to just assert that the objective values are increasing?

Reply:
I don't think these should be fragile. Setting the `PoolSearchMode` to `2` and `PoolSolutions` to 3 instructs Gurobi to find the best 3 solutions. These are not solutions Gurobi happened to come across during the solve. They are actually the best 3.

Reply:
Sounds good.